packages/terra-time-input/src/TimeUtil.js

```diff
@@ -261,8 +261,9 @@ class TimeUtil {
    */
   static getVariantFromLocale(props) {
     const { intl, variant } = props;
+    const locale12Hr = ['en-AU', 'en-CA', 'en-GB', 'en-US', 'en', 'es-US'];
```

Comment:
After thinking about this a little more, instead of creating a list of the locales that support both notations it would be better to list the locales that only support the 24-hour notation. The reason being that any new locales we support in the future should by default honor the variant. If we specifically determined that it should only allow 24-hour notation, then we would add it here. Basically, this should be the _denied_ list instead of the _allowed_ list.

Reply:
Changed f773022